cmd/meradm/server.go

```diff
@@ -97,21 +112,37 @@ func initServer(serviceID string, ipPort string) (*types.RealServer, error) {
 		server.Config.Weight = &wrappers.UInt32Value{Value: uint32(w)}
 	}
 
-	if forward != "" {
-		f, ok := types.ForwardMethod_value[strings.ToUpper(forward)]
+	if forwardMethod != "" {
+		f, ok := types.ForwardMethod_value[strings.ToUpper(forwardMethod)]
 		if !ok {
 			return nil, fmt.Errorf("unrecognized forward method")
 		}
 		server.Config.Forward = types.ForwardMethod(f)
 	}
 
-	return server, nil
+	endpointFlag := cmd.Flag("health-endpoint")
+	if endpointFlag != nil && endpointFlag.Changed {
+		server.HealthCheck.Endpoint = &wrappers.StringValue{Value: healthEndpoint}
+	}
+	if healthPeriod != 0 {
+		server.HealthCheck.Period = ptypes.DurationProto(healthPeriod)
+	}
+	if healthTimeout != 0 {
+		server.HealthCheck.Timeout = ptypes.DurationProto(healthTimeout)
+	}
+	if healthUpThreshold != 0 {
+		server.HealthCheck.UpThreshold = uint32(healthUpThreshold)
```

Comment:
Because uint16 forces the user to provide a valid port. 32 bit would allow the user to specific an invalid port. I thought it was simpler to just encode it in the type of the flag rather than having to add an explicit check.

Reply:
Ah yes, forgot about that.